thundra/wrappers/fastapi/fastapi_wrapper.py

```diff
@@ -0,0 +1,108 @@
+        ThundraTracer.create_instance(AsyncioScopeManager())
+        ExecutionContextManager.set_provider(TracingExecutionContextProvider())
+        self.plugin_context = PluginContext(application_info=self.application_info_provider.get_application_info(),
+                                            request_count=0,
+                                            executor=fastapi_executor,
+                                            api_key=self.api_key)
+
+        self.plugins = wrapper_utils.initialize_plugins(self.plugin_context, disable_trace, disable_metric, disable_log,
+                                                        config=self.config)
+
+        web_wrapper_utils.update_application_info(self.application_info_provider, self.plugin_context.application_info,
+                                                  constants.ClassNames['FASTAPI'])
+
+
+    def before_request(self, request, req_body):
+        execution_context = wrapper_utils.create_execution_context()
+        execution_context.platform_data["request"] = request
+        execution_context.platform_data["request"]["body"] = req_body
+
+        self.plugin_context.request_count += 1
+        self.execute_hook("before:invocation", execution_context)
+
+        return execution_context
+
+
+    def after_request(self, execution_context):
+        self.prepare_and_send_reports_async(execution_context)
+
+    def error_handler(self, error):
+        execution_context = ExecutionContextManager.get()
+        if error:
+            execution_context.error = error
+
+        self.prepare_and_send_reports_async(execution_context)
+        
+        
+
+    def __call__(self, original_func):
+        if hasattr(original_func, "_thundra_wrapped") or ConfigProvider.get(config_names.THUNDRA_DISABLE, False):
+            return original_func
+
+        @wraps(original_func)
+        async def wrapper(*args, **kwargs):
+            request = kwargs.get("request")
+            if request is None or getattr(request, '_thundra_wrapped', False):
+                return await original_func(*args, **kwargs)
+            setattr(request, '_thundra_wrapped', True)
+            try:
+                req_body = request._body if hasattr(request, "_body") else None
+                execution_context = self.before_request(request.scope, req_body)
+            except Exception as e:
+                logger.error('Error during the before part of Thundra: {}'.format(e))
+                return await original_func(*args, **kwargs)
+
+            response = None
+            try:
+                response = await original_func(*args, **kwargs)
+            except Exception as e:
+                try:
+                    error = {
+                        'type': type(e).__name__,
+                        'message': str(e),
+                        'traceback': traceback.format_exc()
+                    }
+                    self.error_handler(error)
```

Comment:
It might be safer to get the execution context directly created by `self.before_request` above and pass it to `self.error_handler` as 2nd argument

Reply:
done.